include/picongpu/particles/manipulators/unary/MaxwellJuettner.hpp

```diff
@@ -0,0 +1,177 @@
+    {
+        /** Functor to modify particle momentum based on temperature for relativistic case
+         * Use Maxwell Juettner Distribution, implementation based on https://doi.org/10.1063/1.4919383
+         * This functor is for the highly-relativistic case.
+         * In this case the added momentum follows the Maxwell-Juettner distribution.
+         *
+         * @tparam T_ValueFunctor pmacc::math::operation::*, binary functor type to
+         *                        add a new momentum to an old one
+         */
+        /** USE THIS IN THE SAME WAY AS Temperature.hpp JUST REPLACE Temperature with MaxwellJuettner
+         * FOR ALL FUNCTOR AND TEMPLATE NAMES**/
+        template<typename T_ValueFunctor>
+        struct MaxwellJuettnerImpl : private T_ValueFunctor
+        {
+            /** Manipulate the momentum of the given macroparticle
+             *
+             * @tparam T_UniformRng functor::misc::RngWrapper, standard
+             *                             normal random number generator type
+             * @tparam T_Particle particle type
+             *
+             * @param uniformRng random number generator
+             * @param particle particle to be manipulated
+             * @param temperatureKeV temperature value in keV
+             */
+            template<typename T_UniformNoZeroRng, typename T_Particle, typename T_MaxwellJuettner>
+            HDINLINE void operator()(
+                T_UniformNoZeroRng& uniformNoZeroRng,
+                T_Particle& particle,
+                T_MaxwellJuettner temperatureKeV) const
+            {
+                float_X const energy = (temperatureKeV * UNITCONV_keV_to_Joule) / UNIT_ENERGY;
+                float_X const macroWeighting = particle[weighting_];
+                float_X const macroMass = attribute::getMass(macroWeighting, particle);
+                float_X const mass = macroMass / macroWeighting;
+                float_X const theta = energy / (mass * SPEED_OF_LIGHT * SPEED_OF_LIGHT);
+
+                float_X x1, x2, x3, x4, u, nu;
+                bool rejected = true;
+                while(rejected)
+                {
+                    x1 = uniformNoZeroRng();
+                    x2 = uniformNoZeroRng();
+                    x3 = uniformNoZeroRng();
+
+                    u = (-1._X) * theta * math::log(x1 * x2 * x3);
+
+                    x4 = uniformNoZeroRng();
+                    nu = (-1._X) * theta * math::log(x1 * x2 * x3 * x4);
+
+                    if(nu * nu - u * u > 1._X)
+                    {
+                        rejected = false;
+                    }
+                }
+
+                float_X momAbs = u * macroWeighting * mass;
+                float_X y1 = uniformNoZeroRng();
+                float_X y2 = uniformNoZeroRng();
+
+                float_X sin_2_pi_y2;
+                float_X cos_2_pi_y2;
+                math::sincos(precisionCast<float_X>(2._X * PI * y2), sin_2_pi_y2, cos_2_pi_y2)
```

Comment:
here, a semicolon is missing 
```suggestion
                math::sincos(precisionCast<float_X>(2._X * PI * y2), sin_2_pi_y2, cos_2_pi_y2);
```